### PR TITLE
chore: Replace deprecated command with environment file

### PR DESCRIPTION
**Pull request by jongwooo**
### SUMMARY

Fixes https://github.com/apache/superset/pull/24063#issuecomment-1970757689
CC @geido

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by dpgaspar** [comment]
./scripts/ci_check_no_file_changes.sh is shell script so the unit test is being skipped here.

```
Run ./scripts/ci_check_no_file_changes.sh python
Searching for changes in python files

CHANGED FILES:
scripts/tag_latest_release.sh

No changes detected... Exiting with SUCCESS code
```

**Comment by dpgaspar** [comment]
> ./scripts/ci_check_no_file_changes.sh is shell script so the unit test is being skipped here.
> 
> ```
> Run ./scripts/ci_check_no_file_changes.sh python
> Searching for changes in python files
> 
> CHANGED FILES:
> scripts/tag_latest_release.sh
> 
> No changes detected... Exiting with SUCCESS code
> ```

here this should help: https://github.com/apache/superset/pull/27303

**Comment by jongwooo** [comment]
> here this should help: https://github.com/apache/superset/pull/27303

@dpgaspar Great catch!

**Comment by dpgaspar** [comment]
ok https://github.com/apache/superset/pull/27303 is merged, can you rebase, so we can catch the failing test please

**Comment by jongwooo** [comment]
> ok https://github.com/apache/superset/pull/27303 is merged, can you rebase, so we can catch the failing test please

@dpgaspar Sure :)

**Comment by dpgaspar** [comment]
ok you caught it!

Change: https://github.com/apache/superset/blob/master/tests/unit_tests/scripts/tag_latest_release_test.py#L47 to:
The tag does_not_exist does not exist. Please use a different tag.\n

**Comment by jongwooo** [comment]
@dpgaspar Should I change line [36](https://github.com/apache/superset/blob/master/tests/unit_tests/scripts/tag_latest_release_test.py#L36) as well?

**Comment by dpgaspar** [comment]
> @dpgaspar Should I change line [36](https://github.com/apache/superset/blob/master/tests/unit_tests/scripts/tag_latest_release_test.py#L36) as well?

probably, test it

**Review by dpgaspar — APPROVED** [review]
Nice! thank you for doing this

File: scripts/tag_latest_release.sh [hunk 1]
```diff
@@ -177,14 +177,14 @@ done
 # Determine the result based on the comparison
 if [[ -z "$compare_result" ]]; then
     echo "Versions are equal"
-    echo "::set-output name=SKIP_TAG::true"
+    echo "SKIP_TAG=true" >> $GITHUB_OUTPUT
 elif [[ "$compare_result" == "greater" ]]; then
     echo "This release tag ${GITHUB_TAG_NAME} is newer than the latest."
-    echo "::set-output name=SKIP_TAG::false"
+    echo "SKIP_TAG=false" >> $GITHUB_OUTPUT
```
Comment (eschutho) [inline scripts/tag_latest_release.sh:183]: @jongwooo Thanks for this fix! I think we may need to update this line [here](https://github.com/preset-io/superset/commit/4b77129cc9344a0e2eaabb54be1c99534ba2a0ad#diff-3cf86f687684a2c690521532de75a831def641338f5a9ec1da8add94714def4cR73) as well from `SKIP_TAG::false` to `SKIP_TAG=false`. Do you mind adding one more PR with this other change?

Comment (jongwooo) [reply scripts/tag_latest_release.sh:183]: @eschutho That'd be great. https://github.com/apache/superset/blob/ad6327db95ba8628e9890e2b2813ae088178d9c1/scripts/build_docker.py#L82 right?

**Pull request merged**